minidump-stackwalk/http_symbol_supplier.cc

```diff
@@ -236,6 +236,20 @@ namespace {
   string JoinURL(CURL* curl, const string& url, const string& sub) {
     return url + "/" + URLEncode(curl, sub);
   }
+
+  string GetCodeIDAndFileQueryString(CURL* curl, const CodeModule* module) {
+      string code_file = PathnameStripper::File(module->code_file());
+      if (code_file.empty()) {
+        return "";
+      }
+
+      string code_id = PathnameStripper::File(module->code_identifier());
+      if (code_id.empty()) {
+        return "";
+      }
+
+      return "?code_file=" + URLEncode(curl, code_file) + "&code_id=" + URLEncode(curl, code_id);
```

Comment:
Yes, but if we have `server_url` set to something like `https://symbols.mozilla.org/?use_private=true&other=config` then we have to parse out the query string so we can append the relative url (like `foo.pdb/abcd/foo.sym`). We don't currently do that, we just blindly append the relative url. Since we don't already have some sort of URL parsing code handy it'd be a lot of work for a feature we don't actually need.